### PR TITLE
Began migration towards jobbergate monorepo

### PR DESCRIPTION
**Pull request by dusktreader**
#### What
Renamed jobbergate-api-fastapi to jobbergate and moved source code to
subdirectory.

#### Why
We are moving jobbergate towards a monorepo for the next generation of jobbergate. We will unify the subprojects (like license manager) and eventually archive the legacy jobbergate-api and jobbergate-web projects.

See ROADMAP.rst for the other steps involved in the process

https://app.clickup.com/t/18022949/JOB-92
---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: ROADMAP.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+This explains the roadmap for migrating from the legacy jobbergate apps to the
+next generation of Jobbergate.
+
+
+Work Items::
+
+    [x] Add capability to dump full json output from legacy jobbergate-cli
+    [x] Rename jobbergate-api-fastapi to jobbergate
+    [x] Move jobbergate-api-fastapi code to sub-project in jobbergate/jobbergate-api
+    [ ] Migrate jobbergate-cli v2 branch to sub-project in jobbergate/jobbergate-cli
```
Comment (heitorPB) [inline ROADMAP.rst:14]: What is the meaning of `sub-project`? A directory? A git submodule?

Comment (dusktreader) [reply ROADMAP.rst:14]: A subdirectory which is a python project unto itself. I considered submodules, but that can just cause a lot more complication than I want to jump into now.

Comment (heitorPB) [reply ROADMAP.rst:14]: Would you mind rewording then to be explicit it is a directory? This might be a cause of confusion in the future.

**Review by heitorPB — COMMENTED** [review]
(for some reason, some of my comments were marked as draft (?) and were not send previously. Sorry for that)

File: ROADMAP.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    [ ] Migrate new additions from legacy jobbergate-cli to jobbergate/jobbergate-cli
+    [ ] Add Dockerfile to jobbergate/jobbergate-cli
+    [ ] Move jobbergate-documentation projet to subproject in jobbergate/jobbergate-doc
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply ROADMAP.rst:17]: The issue here is that jobbergate-documentation is already its own project with dependencies (in pyproject.toml) and build scripts (in Makefile). I want to preserve the existing structure at least for now.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+The `jobbergate-api <jobbergate-api/>`_ is a RESTful API that serves as the back-end for the system.
+It is interacted with via the CLI, though requests can be dispatched to it from any other web
+application.
+
+.. todo::
+
+   Fix the link below after project migration is complete
+
+There jobbergate-api offers `interactive documentation via swagger
```
[the remark this answers is not in the document]
Comment (dusktreader) [reply README.rst:72]: Silly typo where I added a couple of letters to "the" somehow. Updated with a little more in the sentence.

**Review by heitorPB — APPROVED** [review]

**Pull request merged**